### PR TITLE
MS 1455 save return cookie

### PR DESCRIPTION
**Pull request by sulthan-ahmed**
* Write to the db to add a new nrm cookie
* Read a nrm cookie returning the uuid
* Search for a nrm cookie

### CODE REVIEW

File: apps/verify/models/read-cookie.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+const dataService = require('../../common/data-service');
+const {
+makeRequest,
```
Comment (zuzak) [inline apps/verify/models/read-cookie.js]: bad indentation

File: apps/verify/models/write-cookie.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    method: 'post',
+    body: JSON.stringify(body),
+    headers: requestHeaders,
+  };
+
+  const response = await makeRequest('/cookies', options);
+  if (response) {
+    return uuid;
+  }
+  return null;
```
Comment (zuzak) [inline apps/verify/models/write-cookie.js]: Should this throw an error instead?

File: apps/verify/models/read-cookie.js (hunk 1 above)
Comment (sulthan-ahmed) [reply apps/verify/models/read-cookie.js]: good spot copy pasted from elsewhere, I've fixed that as well now 😸

File: apps/verify/models/write-cookie.js (hunk 2 above)
Comment (sulthan-ahmed) [reply apps/verify/models/write-cookie.js]: yes, great work on finding that, highlighted a potential major issue

File: apps/verify/models/cookie.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+const uuidv1 = require('uuid/v1');
+const read = require('./read-cookie');
+const write = require('./write-cookie');
+
+const readUuid = async(uuid) => {
+  return await read(uuid);
+};
+
+const writeCookie = async(data) => {
+  data.uuid = uuidv1();
```
Comment (zuzak) [inline apps/verify/models/cookie.js]: UUIDv1 is an interesting choice -- they're predictable, so if you know one you can predict another.
Consider using v4, which is more properly random.

File: apps/verify/models/read-cookie.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+  const options = {
+    method: 'get',
+    body: null,
+    headers: requestHeaders,
+  };
+
+  try {
+    const response = await makeRequest(`/cookies/${uuid}`, options);
+    if (response) {
```
Comment (zuzak) [inline apps/verify/models/read-cookie.js]: What happens if `response` is falsy?

File: apps/verify/models/cookie.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+const uuidv4 = require('uuid/v4');
+const read = require('./read-cookie');
+const write = require('./write-cookie');
+
+const readUuid = async(uuid) => {
+  return await read(uuid);
+};
+
+const writeCookie = async(data) => {
+  data.uuid = uuidv4();
```
Comment (zuzak) [inline apps/verify/models/cookie.js]: This will overwrite `.uuid` if it already exists. Is that okay?

File: apps/verify/models/read-cookie.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+ * @param {string} uuid - unique id for the cookie
+ *
+ * @returns {Promise} - response from data service (the row data)
+ */
+const readDataFromStore = async(uuid) => {
+
+  const options = {
+    method: 'get',
+    body: null,
+    headers: requestHeaders,
```
Comment (zuzak) [inline apps/verify/models/read-cookie.js]: As `headers` is a common element in all of these, and are just coming straight from `../../common/data-service`, would it be better to just bake the headers into `makeRequest()` itself?

File: apps/verify/models/write-cookie.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+const dataService = require('../../common/data-service');
+const {
+  makeRequest,
+  requestHeaders,
+  } = dataService;
```
Comment (zuzak) [inline apps/verify/models/write-cookie.js]: You don't consume `dataService` directly, so why bother declaring an entire constant?
```suggestion
const {
  makeRequest,
  requestHeaders,
} = require('../../common/data-service');
```

File: apps/verify/models/write-cookie.js [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+ */
+const sendDataToBeStored = async(data) => {
+  // current date
+  const created = (new Date()).toISOString();
+  const uuid = data.uuid;
+  const userEmail = data['user-email'];
+
+  const body = {
+    'created': created,
+    'uuid': uuid,
```
Comment (zuzak) [inline apps/verify/models/write-cookie.js]:
```suggestion
    created,
    uuid,
```

File: apps/verify/models/write-cookie.js [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+    method: 'post',
+    body: JSON.stringify(body),
+    headers: requestHeaders,
+  };
+
+  try {
+    const response = await makeRequest('/cookies', options);
+    if (response) {
+      return uuid;
+    }
```
Comment (zuzak) [inline apps/verify/models/write-cookie.js]: if there's no `response` it'll return `undefined` -- is this ok?

File: apps/verify/models/cookie.js (hunk 5 above)
Comment (sulthan-ahmed) [reply apps/verify/models/cookie.js]: yeah it's OK i plan to create it here

Comment (sulthan-ahmed) [reply apps/verify/models/cookie.js]: it doesn't exist

File: apps/verify/models/read-cookie.js (hunk 6 above)
Comment (sulthan-ahmed) [reply apps/verify/models/read-cookie.js]: copy pasted from Julian tbh

File: apps/verify/models/write-cookie.js (hunk 7 above)
Comment (sulthan-ahmed) [reply apps/verify/models/write-cookie.js]: thanks, I'll just copy and paste so it's in one commit

File: apps/verify/models/read-cookie.js (hunk 6 above)
Comment (sulthan-ahmed) [reply apps/verify/models/read-cookie.js]: I'll fix or @jjrobothamho will fix in another PR

File: apps/verify/models/write-cookie.js (hunk 8 above)
Comment (sulthan-ahmed) [reply apps/verify/models/write-cookie.js]: same will copy and paste so it's in one commit

File: apps/verify/models/write-cookie.js (hunk 9 above)
Comment (sulthan-ahmed) [reply apps/verify/models/write-cookie.js]: see below, does that make sense @zuzak ?

**Comment by sulthan-ahmed** [comment]
I'll merge it as you've gone on holiday @zuzak but feel free to comment on changes for later

**Pull request merged**